zim/plugins/tableofcontents.py

```diff
@@ -76,7 +75,10 @@ def get_headings(parsetree, include_hr):
 		if t[0] == HEADING:
 			level = int(t[1]['level'])
 			text = tokens_to_text(
-						collect_untill_end_token(tokens, HEADING) ).strip()
+				filter_token(
```

Comment:
This fix is already solved another way - please see commit 5b87aa73e3c96772c81dab692a33ef1acbc2c42e